src/ck-ldb/LBDatabase.h

```diff
@@ -197,215 +367,247 @@ class LBDatabase : public IrrGroup {
+  inline void ObjectStop(const LDObjHandle &h) {
+    LBObj* const obj = LbObj(h);
+
+    if (StatsOn()) {
+      LBRealType walltime, cputime;
+      obj->StopTimer(&walltime, &cputime);
+      obj->IncrementTime(walltime, cputime);
+      MeasuredObjTime(walltime, cputime);
+    }
+
+    NoRunningObj();
   };
+  void Send(const LDOMHandle &destOM, const CmiUInt8 &destID, unsigned int bytes, int destObjProc, int force = 0);
+  void MulticastSend(const LDOMHandle &_om, CmiUInt8 *_ids, int _n, unsigned int _b, int _nMsgs=1);
 
   void EstObjLoad(const LDObjHandle &h, double cpuload);
-  inline void NonMigratable(LDObjHandle h) { LDNonMigratable(h); };
-  inline void Migratable(LDObjHandle h) { LDMigratable(h); };
-  inline void setPupSize(LDObjHandle h, size_t pup_size) {LDSetPupSize(h, pup_size);};
-  inline void UseAsyncMigrate(LDObjHandle h, bool flag) { LDAsyncMigrate(h, flag); };
-  inline void DumpDatabase(void) { LDDumpDatabase(myLDHandle); };
+  inline void NonMigratable(LDObjHandle h) { LbObj(h)->SetMigratable(false); };
+  inline void Migratable(LDObjHandle h) { LbObj(h)->SetMigratable(true); };
+  inline void setPupSize(LDObjHandle h, size_t pup_size) { LbObj(h)->setPupSize(pup_size);};
+  inline void UseAsyncMigrate(LDObjHandle h, bool flag) { LbObj(h)->UseAsyncMigrate(flag); };
+  void DumpDatabase(void);
 
   /*
    * Calls from load balancer to load database
    */  
-  inline void NotifyMigrated(LDMigratedFn fn, void *data) 
-  {
-    LDNotifyMigrated(myLDHandle,fn,data);
-  };
- 
-  inline void AddStartLBFn(LDStartLBFn fn, void *data) 
-  {
-    LDAddStartLBFn(myLDHandle,fn,data);
-  };
-
-  inline void RemoveStartLBFn(LDStartLBFn fn) 
-  {
-    LDRemoveStartLBFn(myLDHandle,fn);
-  };
+  int NotifyMigrated(LDMigratedFn fn, void* data);
+  inline void TurnOnNotifyMigrated(int handle)
+       { migrateCBList[handle]->on = 1; }
+  inline void TurnOffNotifyMigrated(int handle)
+       { migrateCBList[handle]->on = 0; }
+  void RemoveNotifyMigrated(int handle);
 
-  inline void StartLB() { LDStartLB(myLDHandle); }
 
-  inline void AddMigrationDoneFn(LDMigrationDoneFn fn, void *data) 
-  {
-    LDAddMigrationDoneFn(myLDHandle,fn,data);
-  };
+  int AddStartLBFn(LDStartLBFn fn, void *data);
+  void TurnOnStartLBFn(int handle)
+       { startLBFnList[handle]->on = 1; }
+  void TurnOffStartLBFn(int handle)
+       { startLBFnList[handle]->on = 0; }
+  void RemoveStartLBFn(LDStartLBFn fn);
 
-  inline void RemoveMigrationDoneFn(LDMigrationDoneFn fn) 
-  {
-    LDRemoveMigrationDoneFn(myLDHandle,fn);
-  };
+  void StartLB();
 
-  inline void MigrationDone() { LDMigrationDone(myLDHandle); }
+  int AddMigrationDoneFn(LDMigrationDoneFn fn, void *data);
+  void RemoveMigrationDoneFn(LDMigrationDoneFn fn);
+  void MigrationDone();
 
 public:
-  inline void TurnManualLBOn() { LDTurnManualLBOn(myLDHandle); }
-  inline void TurnManualLBOff() { LDTurnManualLBOff(myLDHandle); }
+  inline void TurnManualLBOn() { useBarrier = false; }
```

Comment:
Does not seem consistent with what was removed from LBDB. Looks like it may not have been rebased on top of this gerrit change: https://charm.cs.illinois.edu/gerrit/c/charm/+/4999

Reply:
Fixed.